pkl-core/src/main/java/org/pkl/core/module/ModuleKeyFactories.java

```diff
@@ -141,15 +141,20 @@ public Optional<ModuleKey> create(URI uri) {
   private static class File implements ModuleKeyFactory {
     @Override
     public Optional<ModuleKey> create(URI uri) {
-      Path path;
-      try {
-        path = Path.of(uri);
```

Comment:
Needed to adjust this because `Path.of("foo:bar")` throws an IllegalArgumentException, but yet should be treated as a file module key.

Changed this to initialize `Path` within the module key logic rather than here.

Reply:
Yup; this is what we get on Windows:

```
jshell> Path.of(URI.create("file:///foo:bar"))
|  Exception java.nio.file.InvalidPathException: Illegal char <:> at index 4: /foo:bar
|        at WindowsPathParser.normalize (WindowsPathParser.java:182)
|        at WindowsPathParser.parse (WindowsPathParser.java:153)
|        at WindowsPathParser.parse (WindowsPathParser.java:77)
|        at WindowsPath.parse (WindowsPath.java:92)
|        at WindowsUriSupport.fromUri (WindowsUriSupport.java:166)
|        at WindowsFileSystemProvider.getPath (WindowsFileSystemProvider.java:98)
|        at Path.of (Path.java:203)
|        at (#6:1)
```